sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2651,18 +2650,21 @@ def hpat_pandas_series_prod(self, axis=None, skipna=True, level=None, numeric_on
     if not isinstance(self.data.dtype, (types.Integer, types.Float)):
         raise TypingError('{} Non numeric values unsupported. Given: {}'.format(_func_name, self.data.data.dtype))
 
-    if not (isinstance(skipna, (types.Omitted, types.Boolean)) or skipna is True):
+    if not (isinstance(skipna, (types.Omitted, types.Boolean, types.NoneType)) or skipna is None or skipna is True):
         raise TypingError("{} 'skipna' must be a boolean type. Given: {}".format(_func_name, skipna))
 
-    if not (isinstance(axis, types.Omitted) or axis is None) \
-            or not (isinstance(level, types.Omitted) or level is None) \
-            or not (isinstance(numeric_only, types.Omitted) or numeric_only is None) \
-            or not (isinstance(min_count, types.Omitted) or min_count == 0):
+    if not (isinstance(axis, (types.Omitted, types.NoneType)) or axis is None) \
+            or not (isinstance(level, (types.Omitted, types.NoneType)) or level is None) \
+            or not (isinstance(numeric_only, (types.Omitted, types.NoneType)) or numeric_only is None) \
+            or not (isinstance(min_count, (types.Omitted, types.Integer)) or min_count == 0):
         raise TypingError(
             '{} Unsupported parameters. Given axis: {}, level: {}, numeric_only: {}, min_count: {}'.format(
                 _func_name, axis, level, numeric_only, min_count))
 
-    def hpat_pandas_series_prod_impl(self, axis=None, skipna=True, level=None, numeric_only=None, min_count=0):
+    def hpat_pandas_series_prod_impl(self, axis=None, skipna=None, level=None, numeric_only=None, min_count=0):
+        if skipna is None:
+            skipna = True
```

Comment:
This should not compile, since type won't be unified for skipna variable which is None and bool at the same time. You can add a branch based on compile time value, i.e. define skipna_is_none variable at typing and refer to it with:
if skipna_is_none == True:  #noqa
    _skipna = True
else:
    _skipna = skipna